lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_clickhouse_SUITE.erl

```diff
@@ -0,0 +1,329 @@
+    BridgeConfig = clickhouse_config(Config),
+    {ok, _} = emqx_bridge:create(
+        Type,
+        Name,
+        BridgeConfig
+    ),
+    emqx_bridge_resource:bridge_id(Type, Name).
+
+delete_bridge() ->
+    Type = <<"clickhouse">>,
+    Name = atom_to_binary(?MODULE),
+    {ok, _} = emqx_bridge:remove(Type, Name),
+    ok.
+
+reset_table(Config) ->
+    ClickhouseConnection = proplists:get_value(clickhouse_connection, Config),
+    {ok, _, _} = clickhouse:query(ClickhouseConnection, sql_drop_table(), []),
+    {ok, _, _} = clickhouse:query(ClickhouseConnection, sql_create_table(), []),
+    ok.
+
+check_key_in_clickhouse(AttempsLeft, Key, Config) ->
+    ClickhouseConnection = proplists:get_value(clickhouse_connection, Config),
+    check_key_in_clickhouse(AttempsLeft, Key, none, ClickhouseConnection).
+
+check_key_in_clickhouse(Key, Config) ->
+    ClickhouseConnection = proplists:get_value(clickhouse_connection, Config),
+    check_key_in_clickhouse(30, Key, none, ClickhouseConnection).
+
+check_key_in_clickhouse(0, Key, PrevResult, _) ->
+    ct:fail("Expected ~p in database but got ~s", [Key, PrevResult]);
+check_key_in_clickhouse(AttempsLeft, Key, _, ClickhouseConnection) ->
+    {ok, 200, ResultString} = clickhouse:query(ClickhouseConnection, sql_find_key(Key), []),
+    Expected = erlang:integer_to_binary(Key),
+    case iolist_to_binary(string:trim(ResultString)) of
+        Expected ->
+            ok;
+        SomethingElse ->
+            timer:sleep(100),
+            check_key_in_clickhouse(AttempsLeft - 1, Key, SomethingElse, ClickhouseConnection)
+    end.
+
+%%------------------------------------------------------------------------------
+%% Test Cases
+%%------------------------------------------------------------------------------
+
+t_make_delete_bridge(_Config) ->
+    make_bridge(#{}),
+    %% Check that the new brige is in the list of bridges
+    Bridges = emqx_bridge:list(),
+    Name = atom_to_binary(?MODULE),
+    IsRightName =
+        fun
+            (#{name := BName}) when BName =:= Name ->
+                true;
+            (_) ->
+                false
+        end,
+    true = lists:any(IsRightName, Bridges),
+    delete_bridge(),
+    BridgesAfterDelete = emqx_bridge:list(),
+    false = lists:any(IsRightName, BridgesAfterDelete),
+    ok.
+
+t_send_message_query(Config) ->
+    reset_table(Config),
+    BridgeID = make_bridge(#{enable_batch => false}),
+    Key = 42,
+    Payload = #{key => Key, data => <<"clickhouse_data">>, timestamp => 10000},
+    %% This will use the SQL template included in the bridge
+    emqx_bridge:send_message(BridgeID, Payload),
+    %% Check that the data got to the database
+    %%timer:sleep(1000),
+    % {ok, 200, ResultString} = clickhouse:query(ClickhouseConnection, sql_find_key(42), []),
+    % <<"42">> = iolist_to_binary(string:trim(ResultString)),
+
+    check_key_in_clickhouse(Key, Config),
+    delete_bridge(),
+    reset_table(Config),
```

Comment:
Suggestion: we could always delete all bridges and reset the table in `{init,end}_per_testcase` so we avoid inter-testcase dependencies.

Reply:
I now reset table in these hooks. I think it is fine to create and delete the bridge in the cases as the create bridge take different parameters depending on the test case.